### DISCUSSION TITLE
Failed to complete running multiple long-lasting tests when run from JetBrains Rider

### DISCUSSION THREAD

**Discussion by bjkarasek** [MikeSchulze/gdUnit4Mono#159] [Q&A] [answered]
When trying to run many tests from JetBrains Rider where each test takes ~10s to finish, after a few tests succeed, the testing process gets stuck and then a popups shows.
![image](https://github.com/user-attachments/assets/6fc8c747-2ebf-4cf4-983b-52a28594e0df)
After exiting the warning tests are concluded as Aborted or Inconclusive.
![image](https://github.com/user-attachments/assets/ffdf599b-b9fd-4578-9c37-66223f03c0e4)
However when running tests individually there is no issues and they all succeed.

I'm not sure if the issue is on my side. I have tried changing rider settings for Unit Testing but failed to achieve anything.
Also running the tests from Godot Editor works fine, but I wanted to start running them from the Rider since [Exception Hook won't work in Editor](https://github.com/MikeSchulze/gdUnit4Net/discussions/148#discussioncomment-11214056).

I'm working on:
Operating system: Windows 10
<PackageReference Include="gdUnit4.api" Version="4.4.0-rc*"/>
<PackageReference Include="gdUnit4.test.adapter" Version="2.1.0-rc*"/>

Project with repro:
[godot-testing.zip](https://github.com/user-attachments/files/17702554/godot-testing.zip)

**Comment by MikeSchulze** [comment, marked as answer]
Please check the test execution log for more details.
I guess you run in the default session timeout of 30s.
You need to set the session timeout in your `.runsettings`
https://mikeschulze.github.io/gdUnit4/csharp_project_setup/vstest-adapter/#the-test-adapter-settings
```xml
        <!-- set default session timeout to 5m -->
        <TestSessionTimeout>500000</TestSessionTimeout>
```
